index.js

```diff
@@ -45,26 +47,42 @@ var ERROR_TYPES = {
  * This library stores no default configuration of its own. It relies on the application/service
  * it is embedded in to supply this information.
  *
- * @param config
+ * @param appUA an identifying user agent string for your app, compatible with user agent formatting
+ * as per https://tools.ietf.org/html/rfc7231#section-5.5.3 e.g. 'my-app', 'my-app/0.1',
+ * 'my-app/0.1 (Ubuntu/12.04; nodejs/0.10.13)`
+ * @param config object containing config
  * @constructor
  */
-var PersonaClient = function (config) {
-    this.config = config || {};
-
-    var requiredAttributes = [
-        'persona_host', 'persona_port', 'persona_scheme',
-        'persona_oauth_route'
-    ];
-
-    for (var i = 0; i < requiredAttributes.length; i++) {
-        var attribute = requiredAttributes[i];
+var PersonaClient = function (appUA, config) {
+    if (!_.isString(appUA)) {
+        throw new Error("Expected string appId as first parameter");
+    }
 
-        if (this.config[attribute] === undefined) {
-            var name = attribute.replace(/_/g, ' ');
-            throw new Error("You must specify the " + name);
+    // establish config, including defaults
+    this.config = config || {};
+    if (!_.has(this.config,"cert_timeout_sec")) {
+        this.config.cert_timeout_sec = 600;
+    } else {
+        this.config.cert_timeout_sec = parseInt(this.config.cert_timeout_sec,10);
+        if (_.isNaN(this.config.cert_timeout_sec)) {
+            throw new Error("Cert config timeout could not be parsed as integer");
         }
+        // now set refresh interval ms to be equal or just under cert timeout sec
+        this.pk_auto_refresh_timeout_ms =  (this.config.cert_timeout_sec>10) ? (this.config.cert_timeout_sec - 10) * 1000 : this.config.cert_timeout_sec * 1000;
     }
 
+    this.userAgent = (process && _.has(process,["version","env.NODE_ENV"])) ? appUA+
+      " persona-node-client/"+clientVer+" (nodejs/"+process.version+"; NODE_ENV="+
+      process.env.NODE_ENV+")" : appUA + " persona-node-client/"+clientVer;
```

Comment:
does `appUA` need to ensure that it begins with a space as it is being appended with `")"`?